new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java

```diff
@@ -0,0 +1,555 @@
+          .as("Test that createSecret does not throw an exception")
+          .withFailMessage(String.format("Create secret %s failed while creating secret "
+              + "for admin credentials", secretName))
+          .isNotInstanceOf(ApiException.class);
+    }
+  }
+
+  private void createDomain(Domain domain) {
+    boolean result = false;
+    try {
+      result = createDomainCustomResource(domain);
+    } catch (ApiException e) {
+      logger.log(Level.INFO, "createDomainCustomResource failed with ", e);
+      assertThat(e)
+          .as("Test that createDomainCustomResource does not throw an exception")
+          .withFailMessage(String.format(
+              "Could not create domain custom resource for domainUID %s in namespace %s",
+              domainUID, domainNamespace))
+          .isNotInstanceOf(ApiException.class);
+    }
+    assertThat(result)
+        .as("Test createDomainCustomResource returns true")
+        .withFailMessage(String.format(
+            "Create domain custom resource failed for domainUID %s in namespace %s",
+            domainUID, domainNamespace))
+        .isTrue();
+
+  }
+
+  private void checkPodCreated(String podName) {
+    try {
+      waitForCondition(podExists(podName, domainUID, domainNamespace));
+    } catch (ApiException e) {
+      logger.log(Level.INFO, "podExists failed with ", e);
+      assertThat(e)
+          .as("Test that podExists does not throw an exception")
+          .withFailMessage(String.format(
+              "pod %s doesn't exist in namespace %s", podName, domainNamespace))
+          .isNotInstanceOf(ApiException.class);
+    }
+  }
+
+  private void checkPodRunning(String podName) {
+    try {
+      waitForCondition(podReady(podName, domainUID, domainNamespace));
+    } catch (ApiException e) {
+      logger.log(Level.INFO, "podReady failed with ", e);
+      assertThat(e)
+          .as("Test that podReady does not throw an exception")
+          .withFailMessage(String.format(
+              "pod %s is not ready in namespace %s", podName, domainNamespace))
+          .isNotInstanceOf(ApiException.class);
+    }
+  }
+
+  private void checkServiceCreated(String serviceName) {
+    try {
+      waitForCondition(serviceReady(serviceName, null, domainNamespace));
+    } catch (ApiException e) {
+      logger.log(Level.INFO, "podExists failed with ", e);
+      assertThat(e)
+          .as("Test that podExists does not throw an exception")
+          .withFailMessage(String.format(
+              "Service %s is not ready in namespace %s", serviceName, domainNamespace))
+          .isNotInstanceOf(ApiException.class);
+    }
+  }
+
+  private static void waitForCondition(Callable callable) {
+    with().pollDelay(2, SECONDS)
+        .and().with().pollInterval(10, SECONDS)
+        .conditionEvaluationListener(
+            condition -> logger.info(
+                "Waiting for a condition to be met (elapsed time {0}ms, remaining time {1}ms)",
```

Comment:
this message is too generic - the person reading the output will have no idea what you are waiting for

Reply:
this is how the messages look like ....it first logs what's its checking for and then the wait

```
04-14-2020 23:09:36> <INFO> <oracle.weblogic.kubernetes.ItMiiDomain testCreateMiiDomain> <Check for domain custom resouce in namespace ns-ttsf>
<04-14-2020 23:09:38> <INFO> <oracle.weblogic.kubernetes.assertions.impl.Domain lambda$doesDomainExist$0> <Domain Object exists : true>
<04-14-2020 23:09:38> <INFO> <oracle.weblogic.kubernetes.ItMiiDomain lambda$waitForCondition$4> <Waiting for a condition to be met (elapsed time 2,095ms, remaining time 297,905ms)>
<04-14-2020 23:09:38> <INFO> <oracle.weblogic.kubernetes.ItMiiDomain testCreateMiiDomain> <Check for admin server pod domain1-admin-server existence in namespace ns-ttsf>
<04-14-2020 23:09:40> <INFO> <oracle.weblogic.kubernetes.ItMiiDomain lambda$waitForCondition$4> <Waiting for a condition to be met (elapsed time 2,029ms, remaining time 297,971ms)>
<04-14-2020 23:09:50> <INFO> <oracle.weblogic.kubernetes.ItMiiDomain lambda$waitForCondition$4> <Waiting for a condition to be met (elapsed time 12,041ms, remaining time 287,959ms)>
<04-14-2020 23:10:00> <INFO> <oracle.weblogic.kubernetes.ItMiiDomain lambda$waitForCondition$4> <Waiting for a condition to be met (elapsed time 22,062ms, remaining time 277,938ms)>
<04-14-2020 23:10:10> <INFO> <oracle.weblogic.kubernetes.ItMiiDomain lambda$waitForCondition$4> <Waiting for a condition to be met (elapsed time 32,096ms, remaining time 267,904ms)>
<04-14-2020 23:10:20> <INFO> <oracle.weblogic.kubernetes.ItMiiDomain lambda$waitForCondition$4> <Waiting for a condition to be met (elapsed time 42,197ms, remaining time 257,803ms)>
<04-14-2020 23:10:30> <INFO> <oracle.weblogic.kubernetes.ItMiiDomain lambda$waitForCondition$4> <Waiting for a condition to be met (elapsed time 52,230ms, remaining time 247,770ms)>
<04-14-2020 23:10:40> <INFO> <oracle.weblogic.kubernetes.ItMiiDomain lambda$waitForCondition$4> <Waiting for a condition to be met (elapsed time 62,260ms, remaining time 237,740ms)>
<04-14-2020 23:10:51> <INFO> <oracle.weblogic.kubernetes.ItMiiDomain lambda$waitForCondition$4> <Waiting for a condition to be met (elapsed time 72,409ms, remaining time 227,591ms)>
<04-14-2020 23:11:01> <INFO> <oracle.weblogic.kubernetes.ItMiiDomain lambda$waitForCondition$4> <Waiting for a condition to be met (elapsed time 82,486ms, remaining time 217,514ms)>
<04-14-2020 23:11:11> <INFO> <oracle.weblogic.kubernetes.ItMiiDomain lambda$waitForCondition$4> <Waiting for a condition to be met (elapsed time 92,497ms, remaining time 207,503ms)>
<04-14-2020 23:11:21> <INFO> <oracle.weblogic.kubernetes.ItMiiDomain lambda$waitForCondition$4> <Waiting for a condition to be met (elapsed time 102,505ms, remaining time 197,495ms)>
<04-14-2020 23:11:31> <INFO> <oracle.weblogic.kubernetes.assertions.impl.Kubernetes getPod> <Pod Name :domain1-admin-server>
<04-14-2020 23:11:31> <INFO> <oracle.weblogic.kubernetes.assertions.impl.Kubernetes getPod> <Pod Namespace :ns-ttsf>
<04-14-2020 23:11:31> <INFO> <oracle.weblogic.kubernetes.assertions.impl.Kubernetes getPod> <Pod UID :86cc48e9-d7cf-4cd2-8ebf-a739a2059454>
<04-14-2020 23:11:31> <INFO> <oracle.weblogic.kubernetes.assertions.impl.Kubernetes getPod> <Pod Status :Running>
<04-14-2020 23:11:31> <INFO> <oracle.weblogic.kubernetes.ItMiiDomain lambda$waitForCondition$4> <Waiting for a condition to be met (elapsed time 112,523ms, remaining time 187,477ms)>
```

I have the wait as common code, if the wait message is not clear here I can change to have wait for each condition. @rjeberhard @ddsharpe your thoughts please

Reply:
yes but remember that the tests run in parallel, and these will be intermingled with all sorts of other messages, and it seems like a trivial change here to make this much easier for the person reading the output

Reply:
we should not be seeing the intermingled output when tests run in parallel, every test should write to a individual file and that's what we should see to understand what that test is doing. Intermingled output is very confusing and gives wrong impression when looking for a particular test.

Coming to the change, need suggestions. Currently I have this method which is getting called 5 times in this test. I was thinking to add arguments for poll delay, interval, max time to wait. Add one more for logMessage or move this code itself to all 5 places?

```
  private static void waitForCondition(Callable callable) {
    with().pollDelay(2, SECONDS)
        .and().with().pollInterval(10, SECONDS)
        .conditionEvaluationListener(
            condition -> logger.info(
                "Waiting for a condition to be met (elapsed time {0}ms, remaining time {1}ms)",
                condition.getElapsedTimeInMS(),
                condition.getRemainingTimeInMS()))
        // and here we can set the maximum time we are prepared to wait
        .await().atMost(5, MINUTES)
        .until(callable);
  }
```

Reply:
you can create a reusable set of conditions, i have an example, see https://github.com/oracle/weblogic-kubernetes-operator/blob/mark/junit5/new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleDomainValidation.java

Reply:
i am talking about when we run the full suite on jenkins - all the output is intermingled.  i don't really want to dig through 100 different test's output files to work out which one is stuck.
> Intermingled output is very confusing and gives wrong impression when looking for a particular test.
as you said - that is exactly why i am asking you to make this simple change

Reply:
I will make the change. You will never be able to find out correctly what's going on with a test by looking at just the intermingled output.

Reply:
done